docs/yul.rst

```diff
@@ -9,18 +9,19 @@ Yul
 Yul (previously also called JULIA or IULIA) is an intermediate language that can
 compile to various different backends
 (EVM 1.0, EVM 1.5 and eWASM are planned).
-Because of that, it is designed to be a usable common denominator of all three
+Because of this, it is designed to be a usable common denominator of all three
 platforms.
 It can already be used for "inline assembly" inside Solidity and
-future versions of the Solidity compiler will even use Yul as intermediate
+future versions of the Solidity compiler will use Yul as intermediate
 language. It should also be easy to build high-level optimizer stages for Yul.
 
-.. note::
+Yul is implemented when using the compiler in strict assembly mode, which you can use
+by adding the ``--strict-assembly`` and ``--optimize`` arguments when using ``solc``.
+
```

Comment:
I was of the impression that Yul is not yet available to the public for use. Perhaps Chris meant something like "to use the present incarnation of the optimizer, use `solc --strict-assembly --optimize.`

I realize this is confusing terminology which I also had trouble getting my head around :) FWIW, I'm adding my notes below
  - `yul` is the future optimizer (and also a namesake intermediate langugage)
    - the `yul` optimizer supports two intermediate language dialects, namely: `strict assembly` and `yul`
    - `yul` (the intermediate language) is not yet supported but `strict assembly` is
    - people can use the `yul` optimizer by passing the `--strict-assembly --optimize` flag to `solc`. Essentially, they get the benefits of a modern optimizer (`yul`) without having to wait for the release of the namesake intermediate language.

Reply:
I am also somewhat confused as I haven't really been exposed to yul, but this all helps I think. I'll take another pass soon. 👍

Reply:
This is almost correct. We don't really use "Yul" as a term for the optimizer, we usually say "yul optimizer" if we talk about it. Yul is the (intermediate) language. "Using yul" in my mind is taking yul source code, potentially running the optimizer and then generating bytecode output. This bytecode output is not yet implemented for the "strict yul" flavour that has types and a different set of built-in functions. It is fully implemented for the "strict assmebly" flavour which does not have types and uses the EVM instructions as built-in functions.